pykafka/producer.py

```diff
@@ -540,7 +575,40 @@ def flush(self, linger_ms, release_pending=False):
         """
         self._wait_for_flush_ready(linger_ms)
         with self.lock:
-            batch = [self.queue.pop() for _ in range(len(self.queue))]
+            batch = []
+            batch_size_in_bytes = 0
+            while len(self.queue) > 0:
+                # TODO: Should we optimistically pop and renenqueue vs. peak
+                peaked_message = self.queue[-1]
```

Comment:
This might seem pedantic, but I think it's really important for readability. I think you mean "peek" as in "to look at the last item in the queue" rather than "peak" meaning the top of something. Spelling the variable name properly will make it clearer what this code is trying to do.